src/core/qgsrelation.cpp

```diff
@@ -202,6 +204,21 @@ QString QgsRelation::getRelatedFeaturesFilter( const QgsFeature &feature ) const
 {
   QStringList conditions;
 
+  if ( ! d->mPolymorphicRelationId.isEmpty() )
+  {
+    QgsPolymorphicRelation polymorphicRelation = QgsProject::instance()->relationManager()->polymorphicRelation( d->mPolymorphicRelationId );
```

Comment:
I am not very enthusiastic to introduce the dependency on the manager itself, but I am out of better ideas. Storing the `QgsPolymorphicRelation` directly is also not an option.

Reply:
Yeah, I don't like the project instance either, cannot you access the project from `QgsRelationContext` ?

Reply:
Oh, yes, thanks for the input. Nevertheless, I am a bit more concerned that I need to look into the `QgsRelationManager`.